overlord/ifacestate/helpers.go

```diff
@@ -300,6 +325,18 @@ func (m *InterfaceManager) autoConnect(task *state.Task, snapName string, blackl
 			continue
 		}
 		candidates := m.repo.AutoConnectCandidateSlots(snapName, plug.Name, autochecker.check)
+		// If we are in a core transition we may have both the old ubuntu-core
+		// snap and the new core snap providing the same interface. In that
+		// situation we want to ignore any candidates in ubuntu-core and simply
+		// go with those from the new core snap.
+		if len(candidates) == 2 {
+			switch {
+			case candidates[0].Snap.Name() == "ubuntu-core" && candidates[1].Snap.Name() == "core":
+				candidates = candidates[1:2]
+			case candidates[1].Snap.Name() == "ubuntu-core" && candidates[0].Snap.Name() == "core":
+				candidates = candidates[0:1]
+			}
+		}
```

Comment:
as discussed with @mvo it seems once we fixe slot-side auto connect we need something like this for core-support anyway during the transition and as long as we have the transition,  we will have a plug (on core) and two slots (on core and ubuntu-core) and sane autoconnect rules would/should refuse that independent on which side we look at (slots or plugs)